components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -1239,6 +1239,59 @@ GURL JsonRpcService::GetBlockTrackerUrlFromNetwork(std::string chain_id) {
   return GURL();
 }
 
+void JsonRpcService::GetFilEstimateGas(const std::string& from_address,
+                                       const std::string& to_address,
+                                       const std::string& gas_premium,
+                                       const std::string& gas_fee_cap,
+                                       int64_t gas_limit,
+                                       uint64_t nonce,
+                                       const std::string& max_fee,
+                                       const std::string& value,
+                                       GetFilEstimateGasCallback callback) {
+  if (from_address.empty() || to_address.empty()) {
+    std::move(callback).Run(
+        "", "", 0, mojom::FilecoinProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+  auto internal_callback =
+      base::BindOnce(&JsonRpcService::OnGetFilEstimateGas,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+  auto request =
+      fil::getEstimateGas(from_address, to_address, gas_premium, gas_fee_cap,
+                          gas_limit, nonce, max_fee, value);
+  RequestInternal(request, true, network_urls_[mojom::CoinType::FIL],
+                  std::move(internal_callback));
```

Comment:
Please update this based on https://github.com/brave/brave-core/commit/5621c711f0a25203b3b8ed6c48baf7630f54cb4d to run conversion before sanitizer.
And please also update the mocked gas_limit in json_rpc_service_unittest.cc to be INT64_MAX.

Reply:
done